tests/e2e/precompile-upgrade/suites.go

```diff
@@ -0,0 +1,330 @@
+		defaultChainCfgBytes, err := json.Marshal(defaultChainCfg)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// "ewoq" key with "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
+		ewoqPrivKey := "56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027"
+		ewoqKey, err := ethcrypto.HexToECDSA(ewoqPrivKey)
+		gomega.Expect(err).Should(gomega.BeNil())
+		ewoqAddr := ethcrypto.PubkeyToAddress(ewoqKey.PublicKey)
+
+		// "0x4fDdc14F51e0FE9651fcaf081F9ECFA725Ee9af2"
+		privKey2 := "8ac3855ff600db43cf4e9c3a97df1d8ca35d478b8191ecb15c8ed29def82e063"
+		key2, err := ethcrypto.HexToECDSA(privKey2)
+		gomega.Expect(err).Should(gomega.BeNil())
+		addr2 := ethcrypto.PubkeyToAddress(key2.PublicKey)
+
+		blkChainID := ""
+		ginkgo.By("upgrades all nodes with default chain config", func() {
+			utils.Outf("{{magenta}}getting cluster status{{/}}\n")
+			ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+			sresp, err := runnerCli.Status(ctx)
+			cancel()
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			for blkChainID = range sresp.ClusterInfo.CustomChains {
+				utils.Outf("{{magenta}}found block chain ID:{{/}} %q\n", blkChainID)
+				break
+			}
+
+			for _, name := range sresp.ClusterInfo.NodeNames {
+				configPath := filepath.Join(sresp.ClusterInfo.RootDataDir, name, "chainConfigs", blkChainID, "config.json")
+				gomega.Expect(os.MkdirAll(filepath.Dir(configPath), 0777)).Should(gomega.BeNil())
+
+				utils.Outf("{{magenta}}writing chain config for %q{{/}}: %q\n", name, configPath)
+				gomega.Expect(os.WriteFile(configPath, defaultChainCfgBytes, 0777)).Should(gomega.BeNil())
+
+				utils.Outf("{{magenta}}restarting the node %q{{/}}\n", name)
+				ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+				resp, err := runnerCli.RestartNode(ctx, name)
+				cancel()
+				gomega.Expect(err).Should(gomega.BeNil())
+
+				time.Sleep(20 * time.Second)
+
+				ctx, cancel = context.WithTimeout(context.Background(), 15*time.Second)
+				_, err = runnerCli.Health(ctx)
+				cancel()
+				gomega.Expect(err).Should(gomega.BeNil())
+				utils.Outf("{{green}}successfully upgraded %q{{/}} (current info: %+v)\n", name, resp.ClusterInfo.NodeInfos)
+			}
+		})
+
+		evmCli := new(utils.EvmClient)
+		ginkgo.By("makes txs after upgrade", func() {
+			ci := utils.GetClusterInfo()
+			gomega.Expect(len(ci.URIs) > 0).Should(gomega.BeTrue())
+			utils.Outf("{{magenta}}sending txs to subnet-evm endpoints:{{/}} %q\n", ci.SubnetEVMRPCEndpoints)
+
+			ep := ci.SubnetEVMRPCEndpoints[rand.Intn(100)%len(ci.SubnetEVMRPCEndpoints)]
+
+			var err error
+			evmCli, err = utils.NewEvmClient(ep, 25, 1)
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+			prevBal, err := evmCli.FetchBalance(ctx, ewoqAddr)
+			cancel()
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			if prevBal.Cmp(new(big.Int).SetInt64(0)) == 0 {
+				ginkgo.Skip("no balance... skipping tests...")
+			}
```

Comment:
why do we skip the test if the balance is zero? This seems like it could hide an unexpected case

Reply:
Was trying to be generic so it can be run against any cluster. On second thought, will just document instead assuming it will run against local cluster with ewoq wallet having enough balance.

Reply:
Now assume ewoq already has balance and funds addr2